retrieval/discovery/consul/consul.go

```diff
@@ -48,8 +49,40 @@ const (
 	datacenterLabel = model.MetaLabelPrefix + "consul_dc"
 	// serviceIDLabel is the name of the label containing the service ID.
 	serviceIDLabel = model.MetaLabelPrefix + "consul_service_id"
+
+	// Constants for instrumentation.
+	namespace = "prometheus"
 )
 
+var (
+	rpcCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "consul_sd_rpc_calls_total",
+			Help:      "Number of Consul RPC calls.",
+		},
+		[]string{"endpoint", "call"},
+	)
+	rpcFailuresCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "consul_sd_rpc_call_failures_total",
+			Help:      "The number of Consul RPC call failures.",
+		})
+	rpcDuration = prometheus.NewSummary(
+		prometheus.SummaryOpts{
+			Namespace: namespace,
+			Name:      "consul_sd_rpc_call_duration",
+			Help:      "The duration of a Consul RPC call in seconds.",
+		})
+)
+
+func init() {
+	prometheus.MustRegister(rpcCount)
```

Comment:
This should be initilized with the two possible set of label values.

I'd also just drop this and put the labels on the duration metric.